django/db/models/fields/__init__.py

```diff
@@ -678,6 +675,12 @@ def contribute_to_class(self, cls, name, virtual_only=False):
             cls._meta.add_field(self, virtual=True)
         else:
             cls._meta.add_field(self)
+        if self.column:
+            # Refuse to override class methods with the descriptor. This leads to a situation
+            # where if you have a classmethod and a field similarly named, then such fields
+            # can't be deferred (we don't have a check for this yet).
+            if not getattr(cls, self.attname, None):
+                setattr(cls, self.attname, DeferredAttribute(self.attname, cls))
```

Comment:
> So now every access of any field on any Django model always goes through the DeferredAttribute descriptor, correct?

I think this is not the case as `DeferredAttribute` is now a data descriptor (only `__get__` is implemented). The `DeferredAttribute.__get__` method will be called `if 'field' not in instance.__dict__` which is what `Model.__init__` takes care to do `if kwargs['field'] is DEFERRED`.